### DISCUSSION TITLE
umi 路由为什么不支持 pathname + query 传参方式？

### DISCUSSION THREAD

**Discussion by weilun0510** [umijs/umi#9963] [Q&A] [answered]
我记得在react路由里，`pathname + query` 这种方式传参刷新页面参数会丢失。

在使用umi history 的 push 或 replace 时，携带参数跳转到新路由，期望刷新页面时新页面参数会丢失，结果是参数未丢失。
`history.replace('/personal-center', { defaultActiveTab: 'orders' });`
跳到`/personal-center`时，刷新页面，期望`defaultActiveTab`丢失

**Comment by fz6m** [comment, marked as answer]
https://www.appsloveworld.com/reactjs/100/4/how-do-i-clear-location-state-in-react-router-on-page-reload

```ts
  useEffect(() => {
    window.history.replaceState({}, document.title)
  }, [])
```

**Reply by weilun0510** [reply]
谢谢
